tests/functionaltests/UXTestHelpers.mm

```diff
@@ -106,4 +114,27 @@
     }
 }
 
+//
+// UXEvent methods
+//
+void UXEvent::Signal() {
+    [_condition lock];
+
+    _signaled = true;
+    [_condition signal];
+
+    [_condition unlock];
+}
+
+bool UXEvent::Wait(int timeOutInSeconds) {
```

Comment:
>Wait [](start = 14, length = 4)

nit: 

i feel we can change the name to something like TimeoutIn(seconds). returning true when timed out, false otherwise. so the test can be written as 

ASSERT_FALSE_MSG(uxEvent->TimeoutIn(3), "porperty change didn't happen within 3"

current way of naming wait is bit unclear what return value (true/false) stands for
ASSERT_TRUE_MSG(uxEvent->Wait(c_testTimeoutInSec), "FAILED: Waiting for property changed event timed out!");